proof/refine/AARCH64/ArchInvsLemmas_H.thy

```diff
@@ -5,1835 +5,36 @@
-
-lemma idle'_not_queued:
-  "\<lbrakk>valid_idle' s; sym_refs (state_refs_of' s);
-    state_refs_of' s ptr = queue \<times> {rt}\<rbrakk> \<Longrightarrow>
-   ksIdleThread s \<notin> queue"
-  by (frule idle'_no_refs, fastforce simp: valid_idle'_def sym_refs_def)
-
-
-lemma obj_at_conj':
-  "\<lbrakk> obj_at' P p s; obj_at' Q p s \<rbrakk> \<Longrightarrow> obj_at' (\<lambda>k. P k \<and> Q k) p s"
-  by (auto simp: obj_at'_def)
-
-lemma pred_tcb_at_conj':
-  "\<lbrakk> pred_tcb_at' proj P t s; pred_tcb_at' proj Q t s \<rbrakk> \<Longrightarrow> pred_tcb_at' proj (\<lambda>a. P a \<and> Q a) t s"
-  apply (simp add: pred_tcb_at'_def)
-  apply (erule (1) obj_at_conj')
-  done
-
-lemma obj_at_False' [simp]:
-  "obj_at' (\<lambda>k. False) t s = False"
-  by (simp add: obj_at'_def)
-
-lemma pred_tcb_at_False' [simp]:
-  "pred_tcb_at' proj (\<lambda>st. False) t s = False"
-  by (simp add: pred_tcb_at'_def obj_at'_def)
-
-lemma obj_at'_pspaceI:
-  "obj_at' t ref s \<Longrightarrow> ksPSpace s = ksPSpace s' \<Longrightarrow>  obj_at' t ref s'"
-  by (auto intro!: projectKO_stateI simp: obj_at'_def ps_clear_def)
-
-lemma cte_wp_at'_pspaceI:
-  "\<lbrakk>cte_wp_at' P p s; ksPSpace s = ksPSpace s'\<rbrakk> \<Longrightarrow> cte_wp_at' P p s'"
-  apply (clarsimp simp add: cte_wp_at'_def getObject_def)
-  apply (drule equalityD2)
-  apply (clarsimp simp: in_monad loadObject_cte gets_def
-                        get_def bind_def return_def split_def)
-  apply (case_tac b)
-        apply (simp_all add: in_monad typeError_def)
-   prefer 2
-   apply (simp add: in_monad return_def alignError_def assert_opt_def
-                    alignCheck_def magnitudeCheck_def when_def bind_def
-             split: if_split_asm option.splits)
-  apply (clarsimp simp: in_monad return_def alignError_def fail_def assert_opt_def
-                        alignCheck_def bind_def when_def
-                        objBits_cte_conv tcbCTableSlot_def tcbVTableSlot_def
-                        tcbReplySlot_def objBits_defs
-                 split: if_split_asm cong: image_cong
-                 dest!: singleton_in_magnitude_check)
-  done
-
-lemma valid_untyped'_pspaceI:
-  "\<lbrakk>ksPSpace s = ksPSpace s'; valid_untyped' d p n idx s\<rbrakk>
-  \<Longrightarrow> valid_untyped' d p n idx s'"
-  by (simp add: valid_untyped'_def ko_wp_at'_def ps_clear_def)
-
-lemma typ_at'_pspaceI:
-  "typ_at' T p s \<Longrightarrow> ksPSpace s = ksPSpace s' \<Longrightarrow> typ_at' T p s'"
-  by (simp add: typ_at'_def ko_wp_at'_def ps_clear_def)
-
 lemma frame_at'_pspaceI:
   "frame_at' p sz d s \<Longrightarrow> ksPSpace s = ksPSpace s' \<Longrightarrow> frame_at' p sz d s'"
   by (simp add: frame_at'_def typ_at'_def ko_wp_at'_def ps_clear_def)
 
-lemma valid_cap'_pspaceI:
+lemma valid_cap'_pspaceI[Invariants_H_pspaceI_assms]:
   "s \<turnstile>' cap \<Longrightarrow> ksPSpace s = ksPSpace s' \<Longrightarrow> s' \<turnstile>' cap"
```

Comment:
Probably not worth doing and definitely not as part of this PR, but some of these `pspaceI_assms` lemmas feel like they would have the same proof on all architectures and could be generic if the interface was a bit lower down.

Reply:
That's very possible, but it will require effort and I'm thinking it won't save all that many lines. We'll see if any other arch comes into conflict, I'll keep your comment in mind.